crates/rust-analyzer/src/config.rs

```diff
@@ -218,6 +218,10 @@ config_data! {
         /// Advanced option, fully override the command rust-analyzer uses for
         /// formatting.
         rustfmt_overrideCommand: Option<Vec<String>> = "null",
+        /// Enables the use of rustfmt's unstable range formatting command for the
+        /// `textDocument/rangeFormatting` request. The rustfmt option is unstable and only
+        /// available on a nightly build.
+        rustfmt_enableRangeFormatting: bool = "false",
```

Comment:
Seems like we also need to set

https://github.com/rust-analyzer/rust-analyzer/blob/f04daf693aec9f3ffbd98bd368b79646246d506b/crates/rust-analyzer/src/caps.rs#L57

?

Reply:
Yes, fixed.